### PR TITLE
Update to support Zotero 5.0 translator code sharing

### PR DESCRIPTION
**Pull request by adomasven**

### CODE REVIEW

File: src/common/zotero.js [hunk 1]
```diff
@@ -24,17 +24,25 @@
...
+	// http://stackoverflow.com/questions/9847580/how-to-detect-safari-chrome-ie-firefox-and-opera-browser
+	// Firefox 1.0+
+	this.isFirefox = typeof InstallTrigger !== 'undefined';
+	// At least Safari 3+: "[object HTMLElementConstructor]"
+	this.isSafari = Object.prototype.toString.call(window.HTMLElement).indexOf('Constructor') > 0;
+	// Internet Explorer 6-11
+	this.isIE = /*@cc_on!@*/false || !!document.documentMode;
+	// Edge 20+
+	this.isEdge = !this.isIE && !!window.StyleMedia;
+	// Chrome 13+ A deprecated extension function
```
Comment (dstillman) [inline src/common/zotero.js]: Outdated comment?

File: gulpfile.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+});
+
+gulp.task('babelify', babelify);
+gulp.task('minify', minify);
+
+gulp.task('build-dev', sequence('build-sh-dev', 'babelify'));
+
+gulp.task('build', sequence('build-sh', 'babelify', 'minify'));
+
+gulp.task('default', ['build-dev']);
```
Comment (dstillman) [inline gulpfile.js]: Spaces -> tabs

File: gulpfile.js [hunk 3]
```diff
@@ -61,14 +62,24 @@ function reloadChromeExtensionsTab(cb) {
...
         .pipe(plumber({errorHandler: onError}))
         .pipe(babel({
             presets: ['es2015']
         }))
-        .pipe(gulp.dest('./build/safari.safariextension/zotero/'));
+        .pipe(gulp.dest((data) => data.base));
+}
+
+function minify() {
+	return gulp.src('./build/**/*.js')
```
Comment (dstillman) [inline gulpfile.js]: Only the bookmarklet code gets minified, not connector code. (Can't submit minified code to Mozilla.)

Comment (dstillman) [inline gulpfile.js]: These lines also don't work properly if a custom build directory is specified in config.sh. That's OK, but then we should remove that config option and always build to ./build.

File: gulpfile.js [hunk 4]
```diff
@@ -34,88 +34,90 @@ const babel = require('gulp-babel');
...
+	var proc = spawn('./build.sh', [], {cwd: process.cwd()});
+	proc.stdout.on('data', function (data) {
+		process.stdout.write(data);
+	});
+	proc.stderr.on('data', function (data) {
+		process.stderr.write(data);
+	});
+	proc.on('close', function (code) {
+		cb(code)
+	})
```
Comment (dstillman) [inline gulpfile.js]: Most/all of the above should just be squashed into https://github.com/zotero/zotero-connectors/pull/51/commits/b4f2aaa0db3fd5a49374bfe248e3565c0970a76c.

File: build.sh [hunk 5]
```diff
@@ -391,6 +384,11 @@ perl -pe "s|<!--SCRIPTS-->|${scripts:8:$((${#scripts}-10))}|s" "$SRCDIR/safari/I
 | perl -000 -p -e 's|(<key>(?:CFBundleShortVersionString\|CFBundleVersion)</key>\s*)<string>[^<]*</string>|$1<string>'"$VERSION"'</string>|sg' \
 > "$BUILD_DIR/safari.safariextension/Info.plist"
 
+# Transpile Safari JS for Safari 10.0<
+echo "Transpiling Safari JS..." >> "$LOG";
+./node_modules/babel-cli/bin/babel.js "$BUILD_DIR/safari.safariextension/" --out-dir "$BUILD_DIR/safari.safariextension/" --presets es2015 -q >> "$LOG" 2>&1
```
Comment (dstillman) [inline build.sh]: Here and elsewhere, should be an absolute path using `$CWD` (which is badly named, but we can fix that another time). No guarantee that the script will be run from the root.

File: gulpfile.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+});
+
+gulp.task('watch-chrome', function () {
+	var watcher = gulp.watch(['./src/chrome/**', './src/common/**', './src/safari/**']);
+	watcher.on('change', function(event) {
+		gulp.src(event.path)
+			.pipe(processFile())
+			.on('close', reloadChromeExtensionsTab);
+	});
+});
```
Comment (dstillman) [inline gulpfile.js]: Can we just try to do this automatically on macOS, as before, and make it a non-fatal error if chrome-cli fails? A little annoying to have to run a different command.

Comment (adomasven) [inline gulpfile.js]: Yeah, but when you develop for safari you have chrome focusing and refreshing extensions page. Not fun.

Comment (adomasven) [inline gulpfile.js]: Or firefox, edge, later on.

**Pull request merged**